File: examenes/ExamenDialogos/app/src/main/java/com/system/develop/examendialogos/DialgoFecha.java
package com.system.develop.examendialogos;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;

import java.util.Calendar;

public class DialgoFecha extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {


        DatePickerDialog datePickerDialog = new DatePickerDialog(getActivity(), (DatePickerDialog.OnDateSetListener) getActivity()
                ,Calendar.getInstance().get(Calendar.YEAR),
                Calendar.getInstance().get(Calendar.MONTH),
                Calendar.getInstance().get(Calendar.MONTH));

        return datePickerDialog;
    }
}

File: Kotlin/T5/codigos2223/T5_Dialogos/app/src/main/java/com/develop/t5_dialogos/dialogs/DialgoFecha.kt
package com.develop.t5_dialogos.dialogs

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import java.util.Calendar

class DialgoFecha: DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        // builder
        // setear elementos del builder
        // return builder.create

        val calendarioActual: Calendar = Calendar.getInstance();
        val anio= calendarioActual.get(Calendar.YEAR)
        val dia = calendarioActual.get(Calendar.DAY_OF_MONTH)
        val mes =  calendarioActual.get(Calendar.MONTH)

        val dialogoFecha = DatePickerDialog(requireContext(),requireContext() as OnDateSetListener,anio,mes,dia)
        return dialogoFecha
    }
}